proto/pulumi/provider.proto

```diff
@@ -594,35 +603,78 @@ message DiffResponse {
     }
 }
 
+// `CreateRequest` is the type of requests sent as part of a [](pulumirpc.ResourceProvider.Create) call.
 message CreateRequest {
-    string urn = 1;                        // the Pulumi URN for this resource.
-    google.protobuf.Struct properties = 2; // the provider inputs to set during creation.
-    double timeout = 3;                    // the create request timeout represented in seconds.
-    bool preview = 4;                      // true if this is a preview and the provider should not actually create the resource.
-    string name = 5; // the Pulumi name for this resource.
-    string type = 6; // the Pulumi type for this resource.
+    // The URN of the resource being created.
+    string urn = 1;
+
+    // The resource's input properties, to be set during creation. These should have been validated by a call to
+    // [](pulumirpc.ResourceProvider.Check).
+    google.protobuf.Struct properties = 2;
+
+    // A timeout in seconds that the caller is prepared to wait for the operation to complete.
+    double timeout = 3;
+
+    // True if and only if the request is being made as part of a preview/dry run, in which case the provider should not
```

Comment:
I've always been curious what is the point of these preview calls, how are they used? Are they used to generate errors, or they influence the display? Or Both?